src/Button/Button.styles.ts

```diff
@@ -0,0 +1,45 @@
+import styled, { css } from 'styled-components';
+import { Typography } from '../Typography';
+import { SIZES_MAPPING } from './constants';
+import { ButtonVariants, ColorTypes, SizeTypes } from './types';
+import { getButtonVariantStyles, getTypographyVariantStyles } from './utils';
+
+interface StyledButtonProps {
+  $color: ColorTypes;
+  $size: SizeTypes;
+  $variant: ButtonVariants;
+  disabled: boolean;
+}
+
+interface StyledTypographyProps {
+  $color: ColorTypes;
+  $variant: ButtonVariants;
+  disabled: boolean;
+}
+
+const BaseButtonStyles = css<StyledButtonProps>`
+  ${({ theme, $color, $variant, disabled }) => getButtonVariantStyles(theme, $color, $variant, disabled)}
+  align-items: center;
+  display: flex;
```

Comment:
```suggestion
  display: inline-flex;
```
Do you think a Button should be an inline element with flexibility? Giving a `display: flex` to it would make it behave as a block.

Reply:
Makes sense.